CMakeLists.txt

```diff
@@ -52,54 +87,38 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANGXX OR CMAKE_COMPILER_IS_GN
     set(OSMSCOUT_WARNING_FLAGS "${OSMSCOUT_WARNING_FLAGS} -Wno-unused-but-set-parameter")
   endif()
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OSMSCOUT_WARNING_FLAGS}")
+  if(UNIX)
+    add_definitions(-ldl -lm -lpthread)
+  endif()
 endif()
-
-# Enable Large file support
-include(TestLargeFiles)
-OPJ_TEST_LARGE_FILES(OPJ_HAVE_LARGEFILES)
-
+if(APPLE AND NOT ANDROID)
+  set(CMAKE_SIZEOF_VOID_P 4)
```

Comment:
`APPLE` and `ANDROID` are unlikely to occur at the same time, so the `if` statement doesn't seem too useful. And both have 64 bit variants, so forcing `CMAKE_SIZEOF_VOID_P` to 4 seems odd.

If this makes sense, maybe it needs a comment to explain it.

Reply:
Right, that makes no sense.